### PR TITLE
Add new radio, phone and date picker fields to the Form block

### PR DESCRIPTION
**Pull request by EvanHerman**
- Date
- Telephone
- Radio

### CODE REVIEW

**Comment by jrtashjian** [comment]
> Do we have a way to limit this behavior when selecting a new block? Perhaps we only allow a single section featuring all available form blocks?

@paranoia1906 this doesn't look to be possible.

**Comment by jrtashjian** [comment]
I'm resolving the merge conflicts right now

**Comment by jrtashjian** [comment]
Ok, latest changes from master merged in and refactored the new blocks into their own directory structure. Basic save tests added as well.

**Comment by richtabor** [comment]
> Do we have a way to limit this behavior when selecting a new block? Perhaps we only allow a single section featuring all available form blocks?
> 
> formSearchForBlock

Yea this is something from core. I'm not sure if there's an issue yet created for this on the Gutenberg repo or not.

**Comment by EvanHerman** [comment]
> Do we have a way to limit this behavior when selecting a new block? Perhaps we only allow a single section featuring all available form blocks?
> 
> formSearchForBlock

Thought the same thing. It's ugly, but nothing we can do about it (that I could find). Other plugins doing similar things also have the same UI issue.

**Comment by jrtashjian** [comment]
Yeah, I think that if an 'allowedBlocks' prop is set then all allowed blocks should show in a single panel.

File: src/blocks/form/styles/editor.scss [hunk 1]
```diff
@@ -1,4 +1,10 @@
 .editor-block-list__block[data-type="coblocks/form"] {
+	.components-base-control,
```
[the remark this answers is not in the document]
Comment (richtabor) [reply src/blocks/form/styles/editor.scss]: This is _without_ Gutenberg installed.

**Comment by richtabor** [comment]
> On the latest version of this pull request I am getting php errors with specific behavior on the Radio Buttons block.
> 
> **If we drop in a radio buttons block but failed to configure, php errors are thrown**
> image
> 
> **When an option text is provided there are no problems.**
> image
> 
> ```
> Notice: Undefined index: options in /var/www/html/wp-content/plugins/coblocks/includes/class-coblocks-form.php on line 443
> 
> Warning: array_filter() expects parameter 1 to be array, null given in /var/www/html/wp-content/plugins/coblocks/includes/class-coblocks-form.php on line 443
> ```
> 
> Ideally we can handle this situation gracefully. All other Form child blocks are fine without prior configuration.

Looks like your style may not be properly loaded too?

**Comment by richtabor** [comment]
> On the latest version of this pull request I am getting php errors with specific behavior on the Radio Buttons block.
> Notice: Undefined index: options in /var/www/html/wp-content/plugins/coblocks/includes/class-coblocks-form.php on line 443
> 
> Warning: array_filter() expects parameter 1 to be array, null given in /var/www/html/wp-content/plugins/coblocks/includes/class-coblocks-form.php on line 443
> ```
> 
> Ideally we can handle this situation gracefully. All other Form child blocks are fine without prior configuration.

I was able to replicate this as well. @EvanHerman - this is the last bit, after which the PR will be ready for merging. 🤘

**Comment by jrtashjian** [comment]
I think that issue was me. The conditional here just needs to come first:

https://github.com/godaddy/coblocks/pull/969/files#diff-6d80ae73598914e98038e94039011848R443-R449

**Comment by jrtashjian** [comment]
> Looks like your style may not be properly loaded too?

@richtabor Looks like it's working for me.

**Underscores**
underscores

**TwentySeventeen**
twenty-seventeen

**TwentyNineteen**
twenty-nineteen

**Pull request merged**